test2text/pages/reports/report_by_req.py

```diff
@@ -0,0 +1,293 @@
+            query_embedding = embed_requirement(filter_embedding.strip())
+            query_embedding_bytes = serialize_float32(query_embedding)
+            distance_sql = ", vec_distance_L2(embedding, ?) AS distance"
+            distance_order_sql = "distance ASC, "
+
+        where_sql = ""
+        if where_clauses:
+            where_sql = f"WHERE {' AND '.join(where_clauses)}"
+
+    with st.container(border=True):
+        st.session_state.update({"req_form_submitting": True})
+        sql = f"""
+                SELECT
+                    Requirements.id as req_id,
+                    Requirements.external_id as req_external_id,
+                    Requirements.summary as req_summary
+                    {distance_sql}
+                FROM
+                    Requirements
+                {where_sql}
+                ORDER BY
+                    {distance_order_sql}Requirements.id
+                """
+        data = db.conn.execute(
+            sql, params + [query_embedding_bytes] if distance_sql else params
+        )
+        if distance_sql:
+            requirements_dict = {
+                f"#{req_id} Requirement {req_external_id} [smart search d={distance}]": req_id
+                for (req_id, req_external_id, _, distance) in data.fetchall()
+            }
+        else:
+            requirements_dict = {
+                f"#{req_id} Requirement {req_external_id}": req_id
+                for (req_id, req_external_id, _) in data.fetchall()
+            }
+
+        st.subheader("Choose 1 of filtered requirements")
+        option = st.selectbox(
+            "Choose a requirement to work with",
+            requirements_dict.keys(),
+            key="filter_req_id",
+        )
+
+        if option:
+            clause = "Requirements.id = ?"
+            if clause in where_clauses:
+                idx = where_clauses.index(clause)
+                params.insert(idx, requirements_dict[option])
+            else:
+                where_clauses.append(clause)
+                params.append(requirements_dict[option])
+
+            st.subheader("Filter Test cases")
+
+            with st.expander("🔍 Filters"):
+                radius, limit = st.columns(2)
+                with radius:
+                    filter_radius = st.number_input(
+                        "Insert a radius", value=1.00, step=0.01, key="filter_radius"
+                    )
+                    st.info("Max distance to annotation")
+                with limit:
+                    filter_limit = st.number_input(
+                        "Test case limit to show",
+                        min_value=1,
+                        max_value=15,
+                        value=15,
+                        step=1,
+                        key="filter_limit",
+                    )
+                    st.info("Limit of selected test cases")
+
+            if filter_radius:
+                where_clauses.append("distance >= ?")
```

Comment:
Distance should be <=

```suggestion
                where_clauses.append("distance <= ?")
```